From 102e9482907f5b34c7fbea38a057f42352970bef Mon Sep 17 00:00:00 2001
From: Jin Xue <jin.xue@intel.com>
Date: Tue, 22 May 2018 15:00:38 +0800
Subject: [PATCH 4/8] Fix the flicker on single touch

Filter out the dupliate call to seekTo with the same seek time
when seeking with a "touch and leave" manner.

Change-Id: Ie3ba493f38c3cce77b2eb438b59bf544f8e72c2d
Tracked-On: https://jira.devtools.intel.com/browse/OAM-80081
Signed-off-by: Jin Xue <jin.xue@intel.com>
---
 media/libmedia/include/media/mediaplayer.h | 1 +
 media/libmedia/mediaplayer.cpp             | 7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/media/libmedia/include/media/mediaplayer.h b/media/libmedia/include/media/mediaplayer.h
index 2335c5a255..3b235ee749 100644
--- a/media/libmedia/include/media/mediaplayer.h
+++ b/media/libmedia/include/media/mediaplayer.h
@@ -294,6 +294,7 @@ private:
     int                         mCurrentPosition;
     MediaPlayerSeekMode         mCurrentSeekMode;
     int                         mSeekPosition;
+    int                         mSeekTime;
     MediaPlayerSeekMode         mSeekMode;
     bool                        mPrepareSync;
     status_t                    mPrepareStatus;
diff --git a/media/libmedia/mediaplayer.cpp b/media/libmedia/mediaplayer.cpp
index 26908e5570..e53361b831 100644
--- a/media/libmedia/mediaplayer.cpp
+++ b/media/libmedia/mediaplayer.cpp
@@ -60,6 +60,7 @@ MediaPlayer::MediaPlayer()
     mCurrentPosition = -1;
     mCurrentSeekMode = MediaPlayerSeekMode::SEEK_PREVIOUS_SYNC;
     mSeekPosition = -1;
+    mSeekTime = -1;
     mSeekMode = MediaPlayerSeekMode::SEEK_PREVIOUS_SYNC;
     mCurrentState = MEDIA_PLAYER_IDLE;
     mPrepareSync = false;
@@ -584,6 +585,12 @@ status_t MediaPlayer::seekTo_l(int msec, MediaPlayerSeekMode mode)

 status_t MediaPlayer::seekTo(int msec, MediaPlayerSeekMode mode)
 {
+    if (mSeekTime == msec) {
+        ALOGW("Duplicate seek time: %d", msec);
+        return NO_ERROR;
+    } else {
+        mSeekTime = msec;
+    }
     mLockThreadId = getThreadId();
     Mutex::Autolock _l(mLock);
     status_t result = seekTo_l(msec, mode);
--
2.21.0
